docs/roi_tables.md

```diff
@@ -0,0 +1,333 @@
+```json
+{
+    "fractal_roi_table_version": "1",
+    "encoding-type": "anndata",      # Automatically added by anndata
+    "encoding-version": "0.1.0",     # Automatically added by anndata
+}
+```
+
+This is the kind of tables that are used in `fractal-tasks-core` to store ROIs
+like a whole well, or the list of field of views.
+
+#### Single table (segmented objects)
+
+When each table row corresponds to (the bounding box of) a segmented object,
+`fractal-tasks-core` follows more closely the [proposed NGFF update mentioned
+above](https://github.com/ome/ngff/pull/64), with the following additional
+requirements on the Zarr group of a given table:
+
+* Attributes must contain a `type` key, with value `ngff:region_table`.
+* Attributes must contain a `region` key; the corresponding value must be an
+  object with a `path` key and a string value (i.e. the path to the data the
+  table is annotating).
+* Attributes may include a key `instance_key`, which is the key in `obs` that
+  denotes which instance in `region` the row corresponds to. If `instance_key`
+  is not provided, the values from the `_index` Zarr attribute of `obs` is used.
+
+Here is an example of `image.zarr/tables/table1/.zattrs`
+```json
+{
+    "fractal_roi_table_version": "1",
+    "type": "ngff:region_table",
+    "region": {
+        "path": "../labels/label_DAPI",
+    },
+    "instance_key": "label",
+    "encoding-type": "anndata",      # Automatically added by anndata
+    "encoding-version": "0.1.0",     # Automatically added by anndata
+}
+```
+
+### AnnData table format
+
+Data of a table are stored into a Zarr group as AnnData ("Annotated Data")
+objects; the [`anndata` Python library](https://anndata.readthedocs.io) provides the
+definition of this format and the relevant tools.
+
+Quoting from `anndata` documentation:
+
+> AnnData is specifically designed for matrix-like data. By this we mean that
+> we have $n$ observations, each of which can be represented as $d$-dimensional
+> vectors, where each dimension corresponds to a variable or feature. Both the
+> rows and columns of this $n \times d$ matrix are special in the sense that
+> they are indexed.
+>
+> (https://anndata.readthedocs.io/en/latest/tutorials/notebooks/getting-started.html)
+
+Note that AnnData tables are easily transformed from/into `pandas.DataFrame`
+objects - see e.g. the [`AnnData.to_df`
+method](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.to_df.html#anndata.AnnData.to_df).
+
+### Table contents
+
+The [`var` attribute of AnnData
+objects](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.var.html#anndata.AnnData.var)
+indexes the columns of the table. A `fractal-tasks-core` ROI table must include
+the following six columns:
+
+* `x_micrometer`, `y_micrometer`: the lower bounds of the XY intervals defining the ROI, in micrometers;
+* `z_micrometer`: the lower bound of the Z interval defining the ROI, in arbitrary units or in micrometers;
```

Comment:
Oh, I was not aware we're not using the actual units here. That would be a bug, the intent is to use the actual units!